### ISSUE TITLE
Implement multi-task manager

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#79]
We need to modify the structure to allow for training on multiple tasks simultaneously.

**Task**: Any property we want to predict, or other objective we want to accomplish.

## Goal of the implementation
Some properties cannot be combined easily in a single dataset, or cannot use the same loss. We need to decouple them into different tasks, each with its own loss, so that the model can be trained on all tasks concurrently.

## How to do it
It is a bit complicated and requires many changes. In the configuration file, remove the part associated to the data loading from the `datamodule`, and remove the loss and metrics from the `predictor`. Instead, create a _dict_ and its associated class called `task_manager`. An example of such config below.

```
multi_task_manager:
  task_batching: [[task_qm1, task_qm2], [task_prot]]
  tasks:
    task_qm1:
      weight: 1.0
      df_path: "data/qm_properties.csv"
      smiles_col: SMILES
      label_cols: [homo, lumo, energy]
      loss: mse
      metrics_list: ["mae", "pearsonr", "f1 > 0.5"]
      target_nan_mask: ignore-flatten
      task_nn:
        out_dim: 3
        hidden_dims: *middle_dim
        depth: 0
        activation: relu
        last_activation: none
        dropout: 0.2
        normalization: batch_norm
        last_normalization: "none"
        residual_type: simple
    task_qm2:
      weight: 0.5
      df_path: "data/qm_properties.csv"
      smiles_col: SMILES
      label_cols: [is_stable]
      loss: bce
      metrics_list: ["auc", "f1 > 0.5"]
      target_nan_mask: ignore-flatten
      task_nn: ....
    task_prot:
        weight: 3.0
        df_path: "data/protein_binding.csv"
        smiles_col: smile
        label_cols: assayID-*
        loss: mse
        metrics_list: ["mae", "pearsonr", "f1 < 0.5"]
        target_nan_mask: ignore-flatten
        task_nn: ....
```

The predictor handling of the metrics and the loss will have to change to accommodate the multi-tasks.
Some tasks like `task_qm1` and `task_qm2` will be concatenated together, while `task_prot` will be done on a different set of molecules concurrently.

Example of how the tasks classes will be:

# The Task Manager Classes

### Inputs
- metrics_configs: The configs of all metrics used across the tasks
- multitask_configs: The configs of all the tasks

### Role
- Read the data efficiently: When SMILES are similar across datasets, only featurize them once
- Prepare the metrics and loss functions
- Initialize all the `TaskManager`
- Add all the weighted losses together
- Generate a tensorflow and YAML report of the metrics for each task
- Control how the `DataModule` loads molecules from each task

### Inputs
- metrics
- featurized molecules
- task_weight

### Role
- Compute the metrics and loss associated to a specific class
